### DISCUSSION TITLE
Is there a "runbook" or best-practices doc for releasing Xcode builds using Keygen CLI?

### DISCUSSION THREAD

**Discussion by czottmann** [keygen-sh/community#132] [Q&A] [answered]
Hi, I want to use Keygen to handle my app releases (macOS 13+, Xcode 15+). While I have already implemented license creation and validation without any issues to speak of, I'm currently scratching  my head looking at Keygen CLI and the available docs. Maybe I'm missing something but I can't seem to find a higher-level document describing the relationships, e.g. between a release and artifact.

What I have found is the Keygen CLI, but it's documentation feels kinda _"how to draw an owl"_ to me. 😅 So I'm kinda stuck at the moment. For example, what's the right type of artifact in this whole process? The `.app` that Xcode is building, or a DMG file, or a TAR file…? Let's say Xcode has built my app:

1. In which form would I present the Xcode build result, i.e. my `.app`?
2. Which architecture would I pass into `keygen upload --arch` – Xcode creates universal binaries.

Thanks!

**Comment by ezekg** [comment, marked as answer]
In the sidebar of the docs, there's a "distribution" section. [Releases](https://keygen.sh/docs/api/releases/), [artifacts](https://keygen.sh/docs/api/artifacts/), and other distribution-related resources, such as [packages](https://keygen.sh/docs/api/packages/), are covered there. But to give you a tl;dr: a release has many artifacts.

1. Typically, you'd distribute the `dmg` installer and not the uncompressed app bundle. 
2. The `arch` attribute for artifacts is there for organizational purposes, and is not required, so you could set it to `universal` or simply leave it blank since it's optional.

Hope that helps? Let me know.

**Reply by czottmann** [reply]
Thanks! I had seen and read those API docs pages, but again, to me a **high-level** overview would've been helpful, with some broad examples like the one you've just given.

So, yes, thank you, those two pieces of information help a lot! 🤙🏼
